lib/Query/Common/CriterionVisitor/CompositeCriterion.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformSolrSearchEngine\Query\Common\CriterionVisitor;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use EzSystems\EzPlatformSolrSearchEngine\Query\CriterionVisitor;
+
+class CompositeCriterion extends CriterionVisitor
+{
+    public function canVisit(Criterion $criterion): bool
+    {
+        return $criterion instanceof Criterion\CompositeCriterion;
+    }
+
+    public function visit(Criterion $criterion, CriterionVisitor $subVisitor = null): string
```

Comment:
Should this be:
```suggestion
    public function visit(Criterion $criterion, ?CriterionVisitor $subVisitor = null): string
```
or is like that it due to parent compatibility?